### ISSUE TITLE
require hook's handler is executed earlier than expected.

### ISSUE THREAD

**Issue by dpnishant** [elastic/require-in-the-middle#3]
I'm trying to hook the `marko` NPM module but the `exports` object that I'm able to access inside the require handler for `marko` is not the correct one. Can anyone please explain why the two `exports` object appear to be different? Is this a bug or something that is expected?    

The code and the output provided below should be sufficiently self-explanatory.

### Code
```js
var path = require('path')
var hook = require('require-in-the-middle')

hook(['marko'], function (exports, name, basedir) {
  var version = require(path.join(basedir, 'package.json')).version

  console.log('loading %s@%s', name, version)

  exports._version = version

  console.log('\n`exports` Inside\n----------------\n');
  console.log(exports);

  return exports
})



var a = require('marko');

console.log('\n`exports` Outside\n----------------\n');
console.log(a);
```

### Output
```js
loading marko@2.11.1

`exports` Inside
----------------

{ c: [Function: createTemplate], _version: '2.11.1' }

`exports` Outside
----------------

{ c: [Function: createTemplate],
  _version: '2.11.1',
  AsyncWriter: { [Function: AsyncWriter] DEFAULT_TIMEOUT: 10000, enableAsyncStackTrace: [Function] },
  load: [Function: load],
  render: [Function],
  stream: [Function],
  createWriter: [Function],
  helpers: 
   { s: [Function: s],
     fv: [Function: fv],
     f: [Function],
     fl: [Function: fl],
     fp: [Function: fp],
     e: [Function: e],
     ne: [Function: notEmpty],
     x: { [Function: escapeXml] attr: [Function: escapeXmlAttr] },
     xa: [Function: escapeXmlAttr],
     nx: [Function: nx],
     a: [Function],
     as: [Function],
     l: [Function: l],
     r: [Function: r],
     t: [Function: t],
     c: [Function: c],
     i: [Function: i],
     xt: [Function: extend] },
  Template: [Function: Template] }
```

**Comment by rochdev** [comment]
Just got bit by this as well.

@dpnishant Did you ever find a workaround?

**Comment by dpnishant** [comment]
@rochdev No I haven't but I noticed that if I modify the snippet above to something like this:

```javascript
var path = require('path')
var hook = require('require-in-the-middle')

hook(['marko'], function (exports, name, basedir) {
  var version = require(path.join(basedir, 'package.json')).version

  console.log('loading %s@%s', name, version)

  exports._version = version
  
  setTimeout(function () {
  	console.log('\n`exports` Inside\n----------------\n');
  	console.log(exports);
  }, 5000);

  return exports
})



var a = require('marko');

console.log('\n`exports` Outside\n----------------\n');
console.log(a);
```

then the output becomes


```javascript
loading marko@2.11.1

`exports` Outside
----------------

{ c: [Function: createTemplate],
  _version: '2.11.1',
  AsyncWriter: { [Function: AsyncWriter] DEFAULT_TIMEOUT: 10000, enableAsyncStackTrace: [Function] },
  load: [Function: load],
  render: [Function],
  stream: [Function],
  createWriter: [Function],
  helpers: 
   { s: [Function: s],
     fv: [Function: fv],
     f: [Function],
     fl: [Function: fl],
     fp: [Function: fp],
     e: [Function: e],
     ne: [Function: notEmpty],
     x: { [Function: escapeXml] attr: [Function: escapeXmlAttr] },
     xa: [Function: escapeXmlAttr],
     nx: [Function: nx],
     a: [Function],
     as: [Function],
     l: [Function: l],
     r: [Function: r],
     t: [Function: t],
     c: [Function: c],
     i: [Function: i],
     xt: [Function: extend] },
  Template: [Function: Template] }

`exports` Inside
----------------

{ c: [Function: createTemplate],
  _version: '2.11.1',
  AsyncWriter: { [Function: AsyncWriter] DEFAULT_TIMEOUT: 10000, enableAsyncStackTrace: [Function] },
  load: [Function: load],
  render: [Function],
  stream: [Function],
  createWriter: [Function],
  helpers: 
   { s: [Function: s],
     fv: [Function: fv],
     f: [Function],
     fl: [Function: fl],
     fp: [Function: fp],
     e: [Function: e],
     ne: [Function: notEmpty],
     x: { [Function: escapeXml] attr: [Function: escapeXmlAttr] },
     xa: [Function: escapeXmlAttr],
     nx: [Function: nx],
     a: [Function],
     as: [Function],
     l: [Function: l],
     r: [Function: r],
     t: [Function: t],
     c: [Function: c],
     i: [Function: i],
     xt: [Function: extend] },
  Template: [Function: Template] }
```

**Comment by watson** [comment]
This have now been fixed in v2.2.2
